pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -9306,6 +9320,354 @@ spec:
+    tekton.dev/memberOf: tasks
+    tekton.dev/pipeline: p-dag-2
+`),
+			mustParseTaskRunWithObjectMeta(t,
+				taskRunObjectMeta("pr-echo-platforms-1", "foo",
+					"pr", "p-dag-2", "echo-platforms", false),
+				`
+spec:
+  params:
+  - name: platform
+    value: mac
+  serviceAccountName: test-sa
+  taskRef:
+    name: mytask
+    kind: Task
+labels:
+    tekton.dev/memberOf: tasks
+    tekton.dev/pipeline: p-dag-2
+`),
+			mustParseTaskRunWithObjectMeta(t,
+				taskRunObjectMeta("pr-echo-platforms-2", "foo",
+					"pr", "p-dag-2", "echo-platforms", false),
+				`
+spec:
+  params:
+  - name: platform
+    value: windows
+  serviceAccountName: test-sa
+  taskRef:
+    name: mytask
+    kind: Task
+labels:
+    tekton.dev/memberOf: tasks
+    tekton.dev/pipeline: p-dag-2
+`),
+		},
+		expectedPipelineRun: parse.MustParseV1beta1PipelineRun(t, `
+metadata:
+  name: pr
+  namespace: foo
+  annotations: {}
+  labels:
+    tekton.dev/pipeline: p-dag-2
+spec:
+  serviceAccountName: test-sa
+  pipelineRef:
+    name: p-dag-2
+status:
+  pipelineSpec:
+    tasks:
+    - name: pt-with-result
+      params:
+        - name: platforms
+          type: array
+      taskRef:
+        name: taskwithresults
+        kind: Task
+    - name: echo-platforms
+      taskRef:
+        name: mytask
+        kind: Task
+      matrix:
+        params:
+          - name: platform
+            value:
+              - $(tasks.pt-with-result.results.platforms[0])
+              - $(tasks.pt-with-result.results.platforms[1])
+              - $(tasks.pt-with-result.results.platforms[2])
+  conditions:
+  - type: Succeeded
+    status: "Unknown"
+    reason: "Running"
+    message: "Tasks Completed: 1 (Failed: 0, Cancelled 0), Incomplete: 1, Skipped: 0"
+  childReferences:
+  - apiVersion: tekton.dev/v1beta1
+    kind: TaskRun
+    name: pr-pt-with-result
+    pipelineTaskName: pt-with-result
+  - apiVersion: tekton.dev/v1beta1
+    kind: TaskRun
+    name: pr-echo-platforms-0
+    pipelineTaskName: echo-platforms
+  - apiVersion: tekton.dev/v1beta1
+    kind: TaskRun
+    name: pr-echo-platforms-1
+    pipelineTaskName: echo-platforms
+  - apiVersion: tekton.dev/v1beta1
+    kind: TaskRun
+    name: pr-echo-platforms-2
+    pipelineTaskName: echo-platforms
+`),
+	}}
+	for _, tt := range tests {
+		t.Run(tt.pName, func(t *testing.T) {
+			pr := parse.MustParseV1beta1PipelineRun(t, fmt.Sprintf(`
+metadata:
+  name: pr
+  namespace: foo
+spec:
+  serviceAccountName: test-sa
+  pipelineRef:
+    name: %s
+`, tt.pName))
+			d := test.Data{
+				PipelineRuns: []*v1beta1.PipelineRun{pr},
+				Pipelines:    []*v1beta1.Pipeline{tt.p},
+				Tasks:        []*v1beta1.Task{task, taskwithresults},
+				ConfigMaps:   cms,
+			}
+			if tt.tr != nil {
+				d.TaskRuns = []*v1beta1.TaskRun{tt.tr}
+			}
+			prt := newPipelineRunTest(t, d)
+			defer prt.Cancel()
+			pipelineRun, clients := prt.reconcileRun(pr.Namespace, pr.Name /* wantEvents=*/, []string{} /* permanentError=*/, false)
```

Comment:
what's up with the commented out strings in this function?

Reply:
It was suggested by @lbernick  in this comment:     
https://github.com/tektoncd/pipeline/pull/6577#discussion_r1180462200

Reply:
what's the reason for that suggestion?

Reply:
I think the current comment is not exactly the same as Lee suggested? It should be 
` prt.reconcileRun(pr.Namespace, pr.Name, /* wantEvents=*/ []string{}, /* permanentError=*/ false)`

Reply:
Oh it seems on save, it formats it to add commas between `/* wantEvents=*/` `[]string{}` and `/* permanentError=*/` `false`. Maybe safer to just take out the comments?

@lbernick @jerop @Yongxuanzhang

Reply:
I'm looking for existing guidance on this and I can't remember where I've seen this before 🤔 

The intention is to document the arguments you're passing into the function. For example, just looking at the function call here, it's obvious that "pr.Namespace" represents a namespace. But it's not obvious what `[]string{}` or `false` do-- you'd have to look at the function definition to find out, which requires jumping around within the file to be able to fully understand the test you're reading. This is an alternative to creating named variables, e.g. `wantEvents := []string{}`. The review suggestion I made was more around using the return value of the function, but I thought the comments would be a helpful additional clarification.

Yongxuan is correct that the comments are documenting the wrong variables.